packages/ui/src/utils/quote.ts

```diff
@@ -242,3 +243,69 @@ export const appendMetadataToRequest = (
     })
   }
 }
+
+export const getCurrentStep = (steps?: Execute['steps'] | null) => {
+  if (!steps) {
+    return { step: null, stepItem: null }
+  }
+
+  const executableSteps = steps.filter(
+    (step) => step.items && step.items.length > 0
+  )
+
+  const step = executableSteps.find((step) =>
+    step.items.some((item) => item.status === 'incomplete')
+  )
+  const stepItem = step?.items.find((item) => item.status === 'incomplete')
+  return { step, stepItem }
+}
+
+export const getSwapEventData = (
+  details: Execute['details'],
+  steps: Execute['steps'] | null,
+  connector?: Connector
+) => {
+  let operation: string | undefined = details?.operation
+
+  if (operation === 'swap') {
+    const isSameChain =
+      details?.currencyIn?.currency?.chainId ===
+      details?.currencyOut?.currency?.chainId
+    if (isSameChain) {
+      operation = 'same_chain_swap'
+    } else if (
+      details?.currencyIn?.currency?.symbol ===
+      details?.currencyOut?.currency?.symbol
+    ) {
+      operation = 'bridge'
```

Comment:
Don't we get the `operation` from the quote? Is it actually a bridge when the symbol is the same on origin and destination?

Reply:
yeah this was some logic that Jason requested, so he didn't have to do it in posthog.

Reply:
I think so long as the symbols are the same and it's not the same chain we consider that a "bridge"